bin/iotaULTester.js

```diff
@@ -173,7 +174,8 @@ function multipleMeasure(commands) {
                 i: config.deviceId,
                 k: config.apikey,
                 d: values
-            }
+            },
+            responseType: 'text'
         };
 
         request(httpRequest, httpPublishHandler);
```

Comment:
In order to homogenize the way the code make request that expect a text response, the same pattern should be applied across all the code base. There are 2 ways right now:

1) Using the function `requestText()`
2) Passing the parameter option `responseType: 'text'`

The first option encapsulate the option seems make it easier to maintain and modify (like the wrapper concept).

Reply:
Actually that is quite a significant change. At the moment a convenience method called `requestText()` where defined only exists in some of the test helpers. It can't be moved to into each agent, so the only other option would be to `requestText()` it to the exposed functions of the iotagent-node-lib instead. The current `request-shim.js` is using the default `export` which is then exposed on `iotagent-node-lib.js` as a named export resulting in import statements like:

```javascript
const request = require('iotagent-node-lib').request;
```

Now it would be possible to alter `iotagent-node-lib.js` to expose a couple of helper functions instead as `utils.js` does:

```javascript
    const httpOptions = {
        url: options.url,
        method: options.method,
        searchParams: options.searchParams || options.qs,
        headers: options.headers,
        throwHttpErrors: options.throwHttpErrors || false,
        retry: options.retry || 0,
        responseType: options.responseType || 'text',
        body: options.body
    };
    request(httpOptions, callback);
}

exports.request = request; 
exports.requestText = requestText;
exports.requestJson = request; 
```

-  Am I correct in assuming that for consistency you'd want a `requestJson()` as well as another alias just for `request`?
-  Different presents are used on `option` for `requestText()` 
-  Should an entrypoint  called  just `request()` be exposed or not?

Then it would be possible to do the following:

```javascript
const request = require('iotagent-node-lib').requestJson;
const request = require('iotagent-node-lib').requestText;
```

but  that would need to be consistent changes across all the PRs.
 
It also opens the question why is the iotagent-node-lib exposing a convenience method for non-JSON comms if all of its own comms are JSON based?

I mean it could be done, but it results in significant changes across multiple PRs, so I want to confirm this is what you meant before making changes.

Reply:
Also should the exposed functions be called `requestAsJSON()` and `requestAsText()` for clarity?

Reply:
Maybe we haven't explained clearly first time. We refer only to IOTA-UL code. At the preset moment in:

https://github.com/telefonicaid/iotagent-ul/blob/c09e712b389d2beb2fe6ad8eecb99fcc2c9c0505/test/unit/ngsiv2/commandsPolling-test.js#L123-L133

https://github.com/telefonicaid/iotagent-ul/blob/c09e712b389d2beb2fe6ad8eecb99fcc2c9c0505/test/unit/ngsiv2/commandsPolling-test.js#L188-L198

https://github.com/telefonicaid/iotagent-ul/blob/c09e712b389d2beb2fe6ad8eecb99fcc2c9c0505/test/unit/ngsiv2/commandsPolling-test.js#L200-L209

https://github.com/telefonicaid/iotagent-ul/blob/c09e712b389d2beb2fe6ad8eecb99fcc2c9c0505/test/unit/ngsiv2/commandsPolling-test.js#L267-L276

You are using the option `responseType: 'text'` when creating the parameter to pass to the `request()` function, while in:

https://github.com/telefonicaid/iotagent-ul/blob/c09e712b389d2beb2fe6ad8eecb99fcc2c9c0505/test/unit/ngsiv2/httpBindings-test.js#L115

https://github.com/telefonicaid/iotagent-ul/blob/c09e712b389d2beb2fe6ad8eecb99fcc2c9c0505/test/unit/ngsiv2/httpBindings-test.js#L157

you are using the function `requestText` without having the parameter `responseType: 'text'`. They should be the same style (since they are doing the same), for the same of homogeneity.

Reply:
Tests fixed in 33cc7af. `responseText` still exists in `bin/iotaULTester.js` (where the original comment was linked)

Reply:
Could you also update `bin/iotaULTester.js` (and also other references, if they exist), in order to have the same structure across all the code?

Reply:
That is the point I was getting at - `bin/iotaULTester.js` is a simulator and (despite its name) has no access to `/test/utils.js`, therefore can't access `utils.requestText()` - the only way around this would be to add a function to the iotagent-node-lib which seems like overkill.